File: P1-Python/P1-Python-PBG/FactoriaCarretera.py
from FactoriaCarreraYBicicleta import FactoriaCarreraYBicicleta
import random
from BicicletaCarretera import BicicletaCarretera
from CarreraCarretera import CarreraCarretera


class FactoriaCarretera(FactoriaCarreraYBicicleta):
    def crearCarrera(self, participantes):
        #Definir aqui la carrera
        #print('Creando carrera de Carretera')
        carrera = CarreraCarretera()

        for i in range(participantes):
            carrera.bicicletas.append(self.crearBicicleta())

        return carrera


    def crearBicicleta(self):
        #Definir aqui la bicicleta
        #print("Creando bicicleta de Carretera")
        id = str(random.randint(0,1000)).zfill(4)
        return BicicletaCarretera(id)

File: DS-p1-ejercicio1-java/src/Factoria/FactoriaCarretera.java
package src.Factoria;

import src.Bicicleta.Bicicleta;
import src.Bicicleta.BicicletaMontana;
import src.Carrera.Carrera;
import src.Carrera.CarreraMontana;

public class FactoriaCarretera implements FactoriaCarreraBicicleta{
    @Override
    public Carrera crearCarrera(int n_bicicletas) {
        return new CarreraMontana(bicicletas);
    }

    @Override
    public Bicicleta crearBicicleta(int id_bicicleta) {
        return new BicicletaMontana(id_bicicleta);
    }

}
